sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridSerializer.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Threading;
+using Azure.Core;
+
+namespace Azure.Messaging.EventGrid
+{
+    internal class EventGridSerializer : IUtf8JsonSerializable
+    {
+        public object _egEvent;
+        public CancellationToken _cancellationToken;
+
+        public EventGridSerializer(object egEvent, CancellationToken cancellationToken)
+        {
+            _egEvent = egEvent;
+            _cancellationToken = cancellationToken;
+        }
+        public void Write(Utf8JsonWriter writer)
+        {
+            JsonObjectSerializer serializer = new JsonObjectSerializer();
+            var stream = new MemoryStream();
+            serializer.Serialize(stream, _egEvent, typeof(object), _cancellationToken);
```

Comment:
Hmm, I would have expected us to do _egEvent.GetType() for the type here, but it seems to be working fine like this..
I would think this type parameter can be used to restrict serialization to some member of the hierarchy, but maybe when you use object, it just defaults to the type of the instance.

https://github.com/dotnet/runtime/issues/30425#issuecomment-518840850 - I guess this is what is happening.
I guess I would still vote to have _egEvent.GetType() just for the sake of clarity.